WordPress/src/main/java/org/wordpress/android/ui/FilteredRecyclerView.java

```diff
@@ -0,0 +1,327 @@
+        }
+    }
+
+
+    /*
+     * adapter used by the filter spinner
+     */
+    private class SpinnerAdapter extends BaseAdapter {
+        private final FilterCriteria[] mFilterValues;
+        private final LayoutInflater mInflater;
+
+        SpinnerAdapter(Context context, FilterCriteria[] filterValues) {
+            super();
+            mInflater = (LayoutInflater)context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
+            mFilterValues = filterValues;
+        }
+
+        @Override
+        public int getCount() {
+            return (mFilterValues != null ? mFilterValues.length : 0);
+        }
+
+        @Override
+        public Object getItem(int position) {
+            if (position < 0 || position >= getCount())
+                return "";
+            return mFilterValues[position];
+        }
+
+        @Override
+        public long getItemId(int position) {
+            return position;
+        }
+
+        @Override
+        public View getView(int position, View convertView, ViewGroup parent) {
+            final View view;
+            if (convertView == null) {
+                view = mInflater.inflate(R.layout.filter_spinner_item, parent, false);
+            } else {
+                view = convertView;
+            }
+
+            final TextView text = (TextView) view.findViewById(R.id.text);
+            FilterCriteria selectedCriteria = (FilterCriteria)getItem(position);
+            text.setText(selectedCriteria.getLabel());
+            return view;
+        }
+
+        @Override
+        public View getDropDownView(int position, View convertView, ViewGroup parent) {
+            FilterCriteria selectedCriteria = (FilterCriteria)getItem(position);
+            final TagViewHolder holder;
+
+            if (convertView == null) {
+                convertView = mInflater.inflate(R.layout.toolbar_spinner_dropdown_item, parent, false);
+                holder = new TagViewHolder(convertView);
+                convertView.setTag(holder);
+            } else {
+                holder = (TagViewHolder) convertView.getTag();
+            }
+
+            holder.textView.setText(selectedCriteria.getLabel());
+            return convertView;
+        }
+
+        private class TagViewHolder {
+            private final TextView textView;
+            TagViewHolder(View view) {
+                textView = (TextView) view.findViewById(R.id.text);
+            }
+        }
+
+        public int getIndexOfCriteria(FilterCriteria tm) {
+            int pos = -1;
+            if (tm != null && mFilterValues != null){
+                for (int i = 0; i < mFilterValues.length; i++) {
+                    FilterCriteria obj = mFilterValues[i];
+                    if (obj.equals(tm)) {
+                        pos = i;
+                        return pos;
+                    }
+                }
+            }
+            return pos;
+        }
+    }
+
+    public interface Listener {
```

Comment:
The naming here seems confusing. The listener itself is generically called `Listener` but assigned via `setLifecycleListener`. It would be more clear to use consistent naming, for example give it a name like `FilterListener` and assign it via `setFilterListener`.

Some comments explaining the purpose of each listener method would also be very helpful.